lib/x86simdsort.h

```diff
@@ -34,5 +51,19 @@ template <typename T>
 XSS_EXPORT_SYMBOL std::vector<size_t>
 argselect(T *arr, size_t k, size_t arrsize, bool hasnan = false);
 
+// sort an object
+template <typename T, typename F>
+XSS_EXPORT_SYMBOL void object_qsort(T *arr, size_t arrsize, const F key_func)
```

Comment:
This would be more idiomatic in C++ if you did:
```c++
template <typename It, typename F> void object_qsort(It begin, It end, F &&key_func)
{
    using T = typename std::iterator_traits<It>::value_type;
#if __cplusplus >= 201703L
    using R = std::invoke_result_t<F, T>;
#else
    using R = std::result_of_t<F>;
#endif
    std::vector<R> keys;
    keys.reserve(std::distance(begin, end));
    for (auto it = first; it != end; ++it)
        keys.emplace_back(key_func(*it));
```